lib/IDE/SyntaxModel.cpp

```diff
@@ -1080,10 +1080,15 @@ bool ModelASTWalker::handleAttrRanges(ArrayRef<DeclAttributeAndRange> DeclRanges
 
   SmallVector<DeclAttributeAndRange, 4> SortedRanges(DeclRanges.begin(),
                                                      DeclRanges.end());
-  std::sort(SortedRanges.begin(), SortedRanges.end(),
-            [&](DeclAttributeAndRange LHS, DeclAttributeAndRange RHS) {
-    return SM.isBeforeInBuffer(LHS.second.Start, RHS.second.End);
-  });
+  std::sort(
+      SortedRanges.begin(), SortedRanges.end(),
+      [&](DeclAttributeAndRange LHS, DeclAttributeAndRange RHS) {
```

Comment:
Could you add a comment that the comparison is like this because ranges may overlap along with a FIXME to investigate why there are overlapped ranges in the first place ?

Reply:
I believe it's not that there are overlapping ranges, but rather when comparing something against itself. So... a very specific instance of overlapping ranges.

Reply:
I see, so a simpler `return SM.isBeforeInBuffer(LHS.second.Start, RHS.second.Start);` would work as well?

Reply:
Yeah, that's what I changed it to. I didn't realize before that the ranges shouldn't overlap otherwise, but that makes sense give that they're source ranges of attributes.